### ISSUE TITLE
Site Editor: Jetpack AI Assistant options aren't available when editing pages

### ISSUE THREAD

**Issue by hngdny** [Automattic/jetpack#39209]
### Quick summary

Jetpack's AI assistant options are greyed out in the Site Editor, when Assembler theme is enabled.

Initially reported on 8293637-zd-a8c

### Steps to reproduce

1. Enable the Assembler theme
2. Go to 'Appearance → Editor' and open an existing page
3. Expand the AI Assistant section in the settings sidebar toolbar or on the Jetpack topbar icon

### What you expected to happen

The buttons "Improve Title", "Generate feedback" and "Generate Image"  should appear.

### What actually happened

4. The AI Assistant options "Improve Title", "Generate feedback" and "Generate Image" are greyed out

<img width="1126" alt="Screen Shot 2024-06-20 at 14 07 27" src="https://github.com/Automattic/jetpack/assets/70492744/82c8e730-12df-4735-b452-156c0a0e1755">

### Workaround
Users can edit pages directly from the Pages dashboard page instead of the Site Editor

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

**Comment by retnonindya** [comment]
📌 REPRODUCTION RESULTS
- Tested on Atomic – Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
Uncertain because on my end, when I edit a page on Editor, I don't see AI Assistant.

<img width="1718" alt="Screenshot 2024-08-08 at 5 14 34 PM" src="https://github.com/user-attachments/assets/a3810a6c-efd8-4efb-b35e-cc0effbd230a">

Jetpack AI Assistant appears when I edit a page from Pages. 

<img width="1717" alt="Screenshot 2024-08-08 at 5 15 43 PM" src="https://github.com/user-attachments/assets/aaab9764-ba9d-4b1d-bc18-b816a96f9d08">

📌 ACTIONS
- Triaged

**Comment by richtabor** [comment]
I'm not confident this is a theme issue. Does this work on posts?

**Comment by annezazu** [comment]
Agreed. I've updated this after I can replicate on my site using the OnyxPulse theme. @jeherve any clue why this wouldn't be showing up? Tried checking the Jetpack repo for more information but couldn't find anything. If anything, after 6.6 where slots were unified across the editors, I'd expect this to be more consistent.

**Comment by jeherve** [comment]
Moving this to the Jetpack repo, where the AI Assistant feature is being built.

**Comment by jeherve** [comment]
> Go to 'Appearance → Editor'

This is the key here. The AI Assistant feature is currently only available in the block editor. It is not available in the site editor.

While this made sense when the site editor was only used to design the layout of your site, it doesn't make sense when you use the site editor to edit individual posts and pages. This is something we should definitely fix.

cc @vianasw

**Comment by peterschimke** [comment]
@jeherve following up on this? Was the fix deployed for this?

**Comment by jeherve** [comment]
@peterschimke No, we haven't worked on this issue yet. We'll update this issue again once we make progress on this. 👍

**Comment by jeherve** [comment]
Also in #41638
